mbq/client/contrib/permissions.py

```diff
@@ -487,3 +534,30 @@ def get_org_refs_for_permission(
             "get_org_refs_for_permission_completed", person_id, scope, result=result
         )
         return result
+
+    @overload  # noqa: F811
+    def get_persons_with_permission(
+        self, scope: str, org_ref: UUIDType
+    ) -> List[Dict]:
+        ...
+
+    @overload  # noqa: F811
+    def get_persons_with_permission(
+        self, scope: str, org_ref: int, ref_type: RefType
+    ) -> List[Dict]:
+        ...
+
+    def get_persons_with_permission(  # noqa: F811
+        self,
+        scope: str,
+        org_ref: Union[UUIDType, int],
+        ref_type: Optional[RefType] = None,
+    ) -> List[Dict]:
+        if ref_type:
+            return self.os_core_client.fetch_persons_with_permission_for_location(
+                scope, ref_type, int(org_ref)
+            )
+        else:
+            return self.os_core_client.fetch_persons_with_permission(
```

Comment:
please emit an event, perhaps with name `get_persons_with_permission_completed` right before returning in this method

Reply:
I think you also want to be incrementing a collector